src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -180,6 +180,10 @@ void RNN<OutputLayerType, InitializationRuleType>::SinglePredict(
     inputSize = predictors.n_elem / rho;
     prevRho = rho;
   }
+  else if (inputSize == 0)
```

Comment:
I think we should initialize ``prevRho`` with 0, to remove the extra check here.